src/core/server/saved_objects/migrations/core/document_migrator.ts

```diff
@@ -159,10 +159,11 @@ export class DocumentMigrator implements VersionedTransformer {
    */
   constructor({
     typeRegistry,
-    kibanaVersion,
+    kibanaVersion: rawKibanaVersion,
     minimumConvertVersion = DEFAULT_MINIMUM_CONVERT_VERSION,
     log,
   }: DocumentMigratorOptions) {
+    const kibanaVersion = rawKibanaVersion.split('-')[0]; // coerce a semver-like string (x.y.z-SNAPSHOT) or prerelease version (x.y.z-alpha) to a regular semver (x.y.z)
```

Comment:
I first attempted to use the `Semver.coerce` function, which would coerce a prerelease version such as `7.12.0-alpha` to a regular semver `7.12.0`. However, this only appears to work for actual prerelease tags such as `-alpha`, _not_ non-standard tags such as `-SNAPSHOT`. So I used a simple split which will work just as well.